public/static/js/admin.js

```diff
@@ -87,6 +88,28 @@ mciModule.controller('AdminSettingsController', ['$scope','$window', 'mciAdminRe
       return;
     }
 
+    try {
+      $scope.Settings.container_pools.pools = jsyaml.safeLoad($scope.tempContainerPools);
+    } catch(e) {
+      alert("Error parsing container pools yaml: " + e);
+      return;
+    }
+
+    if ($scope.tempContainerPools === null || $scope.tempContainerPools === undefined || $scope.tempContainerPools == "") {
+      $scope.Settings.container_pools.pools = [];
+    }
+
+    // do not save settings if duplicate container pool IDs found
+    var uniqueIds = new Set()
+    for (var i = 0; i < $scope.Settings.container_pools.pools.length; i++) {
+      var p = $scope.Settings.container_pools.pools[i]
+      if (uniqueIds.has(p.id)) {
+        alert("Error saving settings: found duplicate container pool ID: " + p.id);
+        return;
+      }
+      uniqueIds.add(p.id)
+    }
```

Comment:
You could use _.uniq (https://underscorejs.org/#uniq)
```javascript
var pools = $scope.Settings.container_pools
var uniqIDs = _.chain(pools)
  .pluch('id')
  .uniq()
  .value()

pools.length == uniqIDs.length && alert('Scheisse')
```
It doesn't show which id is not unique though...

Reply:
I looked into _.uniq, decided to not use it to be able to isolate the duplicate ID.